### PR TITLE
Improve click handling when pushstate is activated.

### LINKED ISSUE #23: Don't intercept clicks when a modifier key is being pressed
Issued discovered here https://github.com/canjs/bit-docs-html-canjs/pull/205

### PR DESCRIPTION
**Pull request by Macrofig**
- Handle javascript void
- Handle target blank
- Handle metkey click

Resolves #23

### CODE REVIEW

File: can-route-pushstate.js [hunk 1]
```diff
@@ -168,7 +168,24 @@ if (!isFileProtocol && hasPushstate) {
 			// Fix for IE showing blank host, but blank host means current host.
 			var linksHost = node.host || window.location.host;
 
-			if(node.href === "javascript://") {
+			// href has some JS in it, let it run
+			if(node.href.trim().indexOf('javascript') === 0) {
+				try {
+					eval(node.href);
```
Comment (justinbmeyer) [inline can-route-pushstate.js]: if you don't call `ev.preventDefault()` ... which should happen just by returning, won't it do the browser's default behavior?

Also, you might want to check for `javascript:` ... less likely someone has a link like `javascript.com` or something.

File: can-route-pushstate.js [hunk 2]
```diff
@@ -168,7 +168,24 @@ if (!isFileProtocol && hasPushstate) {
...
+				}
+			}
+
+			// Do not push state if target is for blank window
+			if(node.target === '_blank'){
+				return;
+			}
+
+			// Do not push state if meta key was pressed, mimicing standard browser behavior
+			if (e.metaKey) {
```
Comment (chasenlehara) [inline can-route-pushstate.js]: You might need to check for other modifier keys: https://github.com/canjs/bit-docs-html-canjs/pull/205/files#diff-8efd6c52b5a29c345f4c332394b874d5

**Comment by justinbmeyer** [comment]
can-route-pushstate listens on the body ... one idea is to have a plugin "beat" `can-route-pushstate` to the punch.  Listen "lower" than can-route-pushstate, check if `.stopPropagation()` should be called to prevent `can-route-pushstate`'s `.preventDefault()`

**Comment by justinbmeyer** [comment]
- check other libraries that write out `<a>` tags.  Do they do this?

**Comment by chasenlehara** [comment]
Closing in favor of https://github.com/canjs/can-route-pushstate/pull/77

**Pull request closed without merging**